### DISCUSSION TITLE
Multilevel modeling: singular fit in more complex models due to zero variance in site_id random effect?

### DISCUSSION THREAD

**Discussion by carly-gray-19** [now-i-know-my-abcd/docs#92] [Stats & Code Help]
I'm working on longitudinal multilevel models with ABCD using `lmer` from `lme4` and have run into a singular fit issue in some of my models. This is specifically happening in models of cis-gender girls' internalizing symptoms (using raw CBCL scores), but I'm not sure that's critical to my question/issue. What's probably more relevant is that I'm working with a subsample of just 1,383 cis-gender girls for these models, so a much smaller sample size than the full ABCD dataset.

Up until now, I've included random effects to account for the nesting of observations within subjects within families within sites, as has been recommended for mixed-effect models with ABCD. However, when I try to add a fifth fixed effect covariate to my models, the model results in a singular fit. I believe the issue stems from the variance of the `site_id` random effect being equal to 0.000 in this model.

I'm curious to hear if anyone has run into this and/or if you have specific recommendations. I'm thinking of simply removing `site_id` from my models because, aside from this being relevant to the data generation process, it is not theoretically important to my models. However, other options I've read about online include using `blme` to set the residual variance to some small fixed value ([see here](https://bbolker.github.io/mixedmodels-misc/glmmFAQ.html#singular-models-random-effect-variances-estimated-as-zero-or-correlations-estimated-as---1)) or using generalized additive mixture models (i.e., `gamm4`). I've not tried other of these approaches before, so I'm inclined to simply remove the random effect of `site_id.` However, if anyone has a case for the other approaches being much stronger, I would go in that direction. Thanks!

**Comment by angielaird** [comment]
Hi, I'm pasting in an anonymous response:

> I think this is your current design: `... + (1|site_id/family_id/subject_id)`

> Looking at the code for a similar analysis, I only included `familyID` (and excluded `site_id`): `... + (1|family_id)`, which seems to run without issues. I didn't include `subject_id`, because my analysis was not longitudinal. More importantly, I decomposed the predictors into a within- and a between-level component, following recommendations from Enders and Tofighi, which may improve model stability.

> I'm not super familiar with the math here, and I'm afraid to recommend something entirely wrong. But if the variance component for `site_id` is zero, there might not be meaningful variation at the site level in the subsample, so removing the site-level random effect (`... + (1|family_id/subject_id)`) could be acceptable. Alternatively, you could treat the site as a fixed effect (`... + factor(site_id) + (1|family_id/subject_id)`) or separate the random effects (`... + (1|site_id) + (1|family_id) + (1|subject_id)`)?

> These recommendations are also in the article linked in the post: https://bbolker.github.io/mixedmodels-misc/glmmFAQ.html#singular-models-random-effect-variances-estimated-as-zero-or-correlations-estimated-as---1
